src/closure_decorator.rs

```diff
@@ -294,47 +244,79 @@ impl<'a> VisitMut for ClosureDecorator<'a> {
         // )
         *expr = *self.register_mut_ast(&mut Expr::Fn(func.take()), ast);
       }
-      Expr::New(new) if new.args.is_some() => {
+      Expr::New(NewExpr { args, callee, .. }) if args.is_some() => {
         // detect if this looks like a new Proxy
         // e.g. `new Proxy({}, {})`
-        let maybe_proxy_class = match new.callee.as_mut() {
-          Expr::Ident(ident) if &ident.sym == "Proxy" => Some(ident),
-          _ => None,
-        };
+        let maybe_proxy_class = callee.as_mut_ident().filter(|ident| &ident.sym == "Proxy");
 
         if maybe_proxy_class.is_some() {
           let proxy_class = maybe_proxy_class.unwrap();
 
-          let args: Vec<ExprOrSpread> = iter::once(ExprOrSpread {
-            expr: Box::new(Expr::Ident(proxy_class.take())),
-            spread: None,
-          })
-          .chain(
-            new
-              .args
-              .as_mut()
-              .unwrap()
-              .iter_mut()
-              .map(|arg| ExprOrSpread {
-                spread: None,
-                expr: arg.expr.take(),
-              }),
-          )
-          .collect();
+          args
+            .iter_mut()
+            .for_each(|arg| arg.visit_mut_children_with(self));
+
+          let args = args
+            .as_mut()
+            .unwrap()
+            .iter_mut()
+            .map(|arg| ExprOrSpread {
+              spread: None,
+              expr: arg.expr.take(),
+            })
+            .collect();
+
+          proxy_class.visit_mut_children_with(self);
 
           // replace the NewExpr with a call to the `bind` interceptor function
-          // foo.bind(bar)
+          // new Proxy({}, {})
           // =>
-          // bind(foo, bar);
-          *expr = Expr::Call(CallExpr {
-            callee: Callee::Expr(Box::new(Expr::Ident(self.proxy.clone()))),
-            span: new.span.clone(),
-            type_args: None,
-            args: args,
-          });
+          // ("PROXY", ...)
+          *expr = self.wrap_proxy(Box::new(proxy_class.take()), args);
+        } else {
+          expr.visit_mut_children_with(self)
         }
+      }
+      Expr::Call(call) => {
```

Comment:
Did you remove the visit_mut_call_expr handler? I don't think those two can co-exist

Reply:
I did! I was scrambling to figure out how to return a different node type and found this beautiful function :-) the other one is removed.

Reply:
![image](https://user-images.githubusercontent.com/289213/192109478-fa335265-9443-4e3c-a380-701d3ac72b70.png)